src/app/pages/subjects/book-viewer/book-viewer.js

```diff
@@ -4,46 +4,46 @@ import {ActiveElementContextProvider} from '~/contexts/active-element';
 import useSubjectsContext from '../context';
 import BookCover from './book';
 import useSubjectCategoryContext from '~/contexts/subject-category';
+import cn from 'classnames';
 
 function useCategorizedBooks() {
     const {books} = useSubjectsContext();
     const categories = useSubjectCategoryContext();
-    const result = {};
-    const addLabels = () => {
-        for (const category of categories) {
-            if (result[category.cms]) {
-                result[category.cms].label = category.html;
-            }
-        }
-    };
 
-    for (const book of books) {
-        book.subjects.forEach((cmsCategory) => {
-            if (!(cmsCategory in result)) {
-                result[cmsCategory] = [];
-            }
-            if (!result[cmsCategory].includes(book)) {
-                result[cmsCategory].push(book);
+    return React.useMemo(
+        () => {
+            const result = {};
+
+            for (const book of books) {
+                book.subjects.forEach((cmsCategory) => {
+                    if (!(cmsCategory in result)) {
+                        result[cmsCategory] = [];
+                    }
+                    if (!result[cmsCategory].includes(book)) {
+                        result[cmsCategory].push(book);
+                    }
+                });
             }
-        });
-    }
 
-    addLabels();
+            for (const category of categories) {
+                if (result[category.cms]) {
```

Comment:
above on line 19 you used `in` to test for the key, I assumed that was because the value might be 0/false so you needed to test if the key existed.  Probably should do the same here?

Reply:
To be completely correct, I need to test in and typeof.